testdata/project-v3-multigroup/Makefile

```diff
@@ -13,12 +17,20 @@ endif
 
 all: manager
 
-# Run tests
-ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
+# Run tests, and set up envtest if not done already.
+ENVTEST_ASSETS_DIR := testbin
+ENVTEST_SCRIPT_URL := https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh
 test: generate fmt vet manifests
-	mkdir -p ${ENVTEST_ASSETS_DIR}
-	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh
-	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test ./... -coverprofile cover.out
+ifeq (,$(wildcard $(ENVTEST_ASSETS_DIR)/setup-envtest.sh))
+	mkdir -p $(ENVTEST_ASSETS_DIR)
+	curl -sSLo $(ENVTEST_ASSETS_DIR)/setup-envtest.sh $(ENVTEST_SCRIPT_URL)
+endif
+	{ \
+	source $(ENVTEST_ASSETS_DIR)/setup-envtest.sh && \
+	fetch_envtest_tools $(ENVTEST_ASSETS_DIR) && \
+	setup_envtest_env $(PWD)/$(ENVTEST_ASSETS_DIR) && \
+	go test ./... -coverprofile cover.out ; \
+	}
```

Comment:
My main concern with this current solution are:
 
- Users are unable to run the tests to troubleshooting via delve or an IDE since the test will ONLY work if we execute them via the source.   
- The target is complex and not too much unstable which reduces it maintainability
- Its dependency with controller-runtime which make harder we keep it maintained. See that controller-runtime is only a framework and who install/use the bins is ONLY the kubebuilder. So, IMO its implementation should be done in Kubebuilder side. Note that if we want to change something on the steps required to get the binaries we need to bump a new version of the controller-runtime and wait for the next release whcih for example do not allow we do it for v2+ since controller-runtime has breaking changes in the upper version.  

IMO: I prefer the approach done in the PR https://github.com/kubernetes-sigs/kubebuilder/pull/1711 which solves all problems described above. 

c/c @estroz @joelanford @DirectXMan12

Reply:
responded on #1711